app/types/s3_access.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from datetime import UTC, datetime, timedelta
+from io import BytesIO
+
+import boto3
+import botocore
+import botocore.exceptions
+
+from app.types.exceptions import (
+    InvalidS3AccessError,
+    InvalidS3BucketNameError,
+    InvalidS3FileNameError,
+)
+
+
+class S3Access:
+    """Class to manage S3 access with object locking."""
+
+    def __init__(
+        self,
+        failure_logger: str,
+        folder: str,
+        s3_bucket_name: str | None = None,
+        s3_access_key_id: str | None = None,
+        s3_secret_access_key: str | None = None,
+        retention: int = -1,
+    ) -> None:
+        self.bucket_name = s3_bucket_name or "default-bucket-name"

```

Comment:
I would suggest to fail if no bucket name is provided, to be sure the logs are not created in an unexpected bucket
```suggestion
        self.bucket_name = s3_bucket_name
```